source-map.bs

```diff
@@ -256,14 +364,166 @@ with four fields represent mapped code where a corresponding name does not exist
 fields represent mapped code that also has a mapped name.
 
 Note: Using file offsets was considered but rejected in favor of using line/column data to avoid becoming
-misaligned with the original due to platform-specific line endings. 
+misaligned with the original due to platform-specific line endings.
+
+A <dfn>decoded mapping</dfn> is a [=struct=] with the following fields:
+<dl dfn-for="decoded mapping">
+  <dt><dfn>generatedLine</dfn></dt>
+  <dd>A non-negative integer.</dd>
+
+  <dt><dfn>generatedColumn</dfn></dt>
+  <dd>A non-negative integer.</dd>
+
+  <dt><dfn>originalSource</dfn></dt>
+  <dd>A [=decoded source=] or null.</dd>
+
+  <dt><dfn>originalLine</dfn></dt>
+  <dd>A non-negative integer or null.</dd>
+
+  <dt><dfn>originalColumn</dfn></dt>
+  <dd>A non-negative integer or null.</dd>
+
+  <dt><dfn>name</dfn></dt>
+  <dd>A [=string=] or null.</dd>
+</dl>
+
+
+To <dfn>decode source map mappings</dfn> given a [=string=]
+<dfn for="decode source map mappings">|mappings|</dfn>, a [=list=] of [=strings=]
+<dfn for="decode source map mappings">|names|</dfn>, and a [=list=] of [=decoded source|decoded
+sources=] <dfn for="decode source map mappings">|sources|</dfn>, run the following steps:
+1. If |mappings| is not an [=ASCII string=], throw an error.
+1. If |mappings| contains any [=code unit=] other than:
+    - U+002C (,) or U+003B (;);
+    - U+0030 (0) to U+0039 (9);
+    - U+0041 (A) to U+005A (Z);
+    - U+0061 (a) to U+007A (z);
+    - U+002B (+), U+002F (/)
+    <!-- TODO: Is = allowed? -->
+
+    NOTE: These are the valid [[base64]] characters (excluding the padding character `=`), together
+    with `,` and `;`.
+
+    then throw an error.
+1. Let |decodedMappings| be a new empty [=list=].
+1. Let |groups| be the result of [=strictly split|strictly splitting=] |mappings| on `;`.
+1. Let |generatedLine| be 0.
+1. While |generatedLine| is less than |groups|'s [=list/size=]:
+    1. If |groups|[|generatedLine|] is not the empty string, then:
+        1. Let |segments| be the result of [=strictly split|strictly splitting=]
+            |groups|[|generatedLine|] on `,`.
+        1. Let |generatedColumn| be 0.
+        1. Let |sourceIndex| be 0.
```

Comment:
Should `|sourceIndex|` through `|nameIndex|` be defined outside of this loop? I believe these should retain their previous value even when the `generatedLine` goes to the next line.